### DISCUSSION TITLE
Credentials provider, redirect after sign in based on user object

### DISCUSSION THREAD

**Discussion by nickcanarelli** [nextauthjs/next-auth#9780] [Help]
### Question 💬

I am trying to redirect a user to a specified slug after the user has successfully logged in however when using the `signIn` callback this negates setting the session cookie as to be expected per the docs.

> Redirects returned by this callback cancel the authentication flow. Only redirect to error pages that, for example, tell the user why they're not allowed to sign in.
> 
> To redirect to a page after a successful sign in, please use the [callbackUrl](https://authjs.dev/reference/utilities#specifying-a-callbackurl) option or [the redirect callback](https://authjs.dev/reference/configuration/auth-config#callbacks).

Is there a way to return the slug from the user in the response from signin?

**This does not work**
```
async signIn({ user }) {
      const slug = user.businessSlug;
      if (!slug) {
        return false;
      } else {
        return 'http://' + slug + '.project.com:4000';
      }
    },
```

Is there any way to get it in this response?
```
await signIn('credentials', {
      email,
      password,
      redirect: false,
    }).then((res) => {
      setIsLoggingIn(false);
      if (!res?.error) {
        // success, do something with the response
        router.push(callbackUrl ? callbackUrl : res.url);
      }
      if (res.error) {
        console.log(res.error);
      }
    });
```

### How to reproduce ☕️

1. Create next app
2. Setup next auth
3. add signin callback to create callbackUrl
4. try logging in

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by sangosteve** [comment]
Hi. Did you find a solution to this?

**Comment by AmandineTournay** [comment]
Hi !
I am having the same thing as I redirect the user depending of its status through the `signIn` callback, but it considers the result as `false` as I get nothing in the `authorized` callback neither `req.auth` inside my middleware file 😢.

**Comment by sshuvoo** [comment]
Almost two years, and the issue is probably still unsolved. 🥲
